libsolidity/ast/AST.cpp

```diff
@@ -181,6 +181,13 @@ void ContractDefinition::setUserDocumentation(Json::Value const& _userDocumentat
 }
 
 
+std::string ContractDefinition::fullyQualifiedName() const
+{
+	std::string sourceString = *(location().sourceName);
```

Comment:
Will look into this.  FYI, the reason I dropped the : when the source unit name is not present is because it breaks a bunch of tests that expect to look up contracts by their name.  It's really just busy work to fix, but I also opted for the path-of-least-edits for a first go-round.